CppCoreGuidelines.md

```diff
@@ -9151,7 +9151,37 @@ be able to destroy a cyclic structure.
 
 ##### Example
 
-    ???
+    #include <memory>
+
+    class bar;
+
+    class foo
+    {
+    public:
+      foo(const std::shared_ptr<bar>& b)
+        : forward_reference{b}
+      { }
+    private:
+      std::shared_ptr<bar> forward_reference;
+    };
+
+    class bar
+    {
+    public:
+      void set_back_reference(const std::weak_ptr<foo>& f)
+      {
+        this->back_reference = f;
+      }
+      void do_something()
+      {
+        std::shared_ptr<foo> shared_back_reference = this->back_reference.lock();
+        if (shared_back_reference) {
```

Comment:
`if(auto shared_back_reference = back_reference.lock())`, per ES.6: Declare names in for-statement initializers and conditions to limit scope